hedera-mirror-web3/src/main/java/com/hedera/modularized/state/AccountReadableKVState.java

```diff
@@ -0,0 +1,277 @@
+import com.hedera.mirror.web3.repository.NftAllowanceRepository;
+import com.hedera.mirror.web3.repository.NftRepository;
+import com.hedera.mirror.web3.repository.TokenAccountRepository;
+import com.hedera.mirror.web3.repository.TokenAllowanceRepository;
+import com.hedera.pbj.runtime.OneOf;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.state.spi.ReadableKVStateBase;
+import jakarta.inject.Named;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.extern.java.Log;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This class serves as a repository layer between hedera app services read only state and the Postgres database in mirror-node
+ *
+ * The object, which is read from DB is converted to the PBJ generated format, so that it can properly be utilized by the hedera app components
+ * */
+@Named
+@Log
+public class AccountReadableKVState extends ReadableKVStateBase<AccountID, Account> {
+    private static final long DEFAULT_AUTO_RENEW_PERIOD = 7776000L;
+    private static final String KEY = "ACCOUNTS";
+    private static final Long ZERO_BALANCE = 0L;
+
+    private final CommonEntityAccessor commonEntityAccessor;
+    private final NftAllowanceRepository nftAllowanceRepository;
+    private final NftRepository nftRepository;
+    private final TokenAllowanceRepository tokenAllowanceRepository;
+    private final CryptoAllowanceRepository cryptoAllowanceRepository;
+    private final TokenAccountRepository tokenAccountRepository;
+    private final AccountBalanceRepository accountBalanceRepository;
+
+    public AccountReadableKVState(
+            CommonEntityAccessor commonEntityAccessor,
+            NftAllowanceRepository nftAllowanceRepository,
+            NftRepository nftRepository,
+            TokenAllowanceRepository tokenAllowanceRepository,
+            CryptoAllowanceRepository cryptoAllowanceRepository,
+            TokenAccountRepository tokenAccountRepository,
+            AccountBalanceRepository accountBalanceRepository) {
+        super(KEY);
+        this.commonEntityAccessor = commonEntityAccessor;
+        this.nftAllowanceRepository = nftAllowanceRepository;
+        this.nftRepository = nftRepository;
+        this.tokenAllowanceRepository = tokenAllowanceRepository;
+        this.cryptoAllowanceRepository = cryptoAllowanceRepository;
+        this.tokenAccountRepository = tokenAccountRepository;
+        this.accountBalanceRepository = accountBalanceRepository;
+    }
+
+    @Override
+    protected Account readFromDataSource(@NotNull AccountID key) {
+        final var timestampValue = ContractCallContext.get().getTimestamp();
+        final Optional<Long> timestamp = timestampValue != 0 ? Optional.of(timestampValue) : Optional.empty();
```

Comment:
It's weird that the two uses of the context timestamp in the codebase is to wrap it into an optional. Hence we should wrap this into an optional once directly in `ContractCallContext.setTimestamp()`.
```suggestion
        final var timestamp = ContractCallContext.get().getTimestamp();
```